apps/fortuna/src/command/inspect.rs

```diff
@@ -0,0 +1,113 @@
+use {
+    crate::{
+        chain::ethereum::{
+            PythContract,
+            Request,
+        },
+        config::{
+            Config,
+            EthereumConfig,
+            InspectOptions,
+        },
+    },
+    anyhow::Result,
+    ethers::{
+        contract::Multicall,
+        middleware::Middleware,
+        prelude::{
+            Http,
+            Provider,
+        },
+    },
+};
+
+pub async fn inspect(opts: &InspectOptions) -> Result<()> {
+    match opts.chain_id.clone() {
+        Some(chain_id) => {
+            let chain_config = &Config::load(&opts.config.config)?.get_chain_config(&chain_id)?;
+            inspect_chain(chain_config, opts.num_requests).await?;
+        }
+        None => {
+            let config = Config::load(&opts.config.config)?;
+            for (chain_id, chain_config) in config.chains.iter() {
+                println!("Inspecting chain: {}", chain_id);
+                inspect_chain(chain_config, opts.num_requests).await?;
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn inspect_chain(chain_config: &EthereumConfig, num_requests: u64) -> Result<()> {
+    let rpc_provider = Provider::<Http>::try_from(&chain_config.geth_rpc_addr)?;
+    let multicall_exists = rpc_provider
+        .get_code(ethers::contract::MULTICALL_ADDRESS, None)
+        .await
+        .expect("Failed to get code")
+        .len()
+        > 0;
+
+    let contract = PythContract::from_config(chain_config)?;
+    let entropy_provider = contract.get_default_provider().call().await?;
+    let provider_info = contract.get_provider_info(entropy_provider).call().await?;
+    let mut current_request_number = provider_info.sequence_number;
+    println!("Initial request number: {}", current_request_number);
+    let last_request_number = current_request_number.saturating_sub(num_requests);
+    if multicall_exists {
+        println!("Using multicall");
+        let mut multicall = Multicall::new(
+            rpc_provider.clone(),
+            Some(ethers::contract::MULTICALL_ADDRESS),
+        )
+        .await?;
+        let batch_size = 100;
```

Comment:
suggest putting this in InspectOptions with a default value, just in case it needs to be changed later.